server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -257,7 +257,7 @@ private Properties readStartupMessage(ByteBuf buffer) {
             }
             String value = readCString(buffer);
             LOGGER.trace("payload: key={} value={}", key, value);
-            if (!"".equals(key) && !"".equals(value)) {
+            if (!key.isEmpty() && !"".equals(value)) {
```

Comment:
Note that `"".equals(key)` does handle `null` key values while `key.isEmpty()` could raise a null pointer exception.

Reply:
yep, that's why I changed only this, not the value, since the null check for `key` happens before and exits the loop.